### PR TITLE
Filter challenges by type

### PR DESCRIPTION
**Pull request by viktornoskire**
When onsite challenges box is checked, only onsite challenges are visible:
onsite challenges

When online challenges box is checked, only online challenges are visible:
online challenges

Changed index.js script tag to type="module" in index.html and challenges.html so that the JS file could be able to take Import/Export.

### CODE REVIEW

**Comment by viktornoskire** [comment]
Also I should add that if both boxes are checked, all challenges are visible. And if no boxes are checked, all challenges are visible as well.

**Comment by Nishune** [comment]
Looks good and great work, i saw your comment about having all challenges visible with both checkboxes filled. I was thinking about the message that should show if no challenges are visible aswell. But i think the user experience is better if all challenges are shown if you have none of the checkboxes "checked". So maybe the checkbox filter just should show all the challenges as they do now in your version. And that we can only filter by online / onsite? 

In other words, i think this is good! What do you think?

**Comment by viktornoskire** [comment]
I think this makes sense and is user friendly. It would maybe be confusing if no challenges were visible unless you check one of the boxes.

**Comment by nilsishome** [comment]
I agree with Rikard, I looked in your code and your image examples… it looks good and looks ready for implementation!

**Comment by Nishune** [comment]
I agree with both of you, i think this is solid.
